lib/pure/collections/sequtils.nim

```diff
@@ -601,12 +603,12 @@ template filterIt*(s, pred: untyped): untyped =
   result
```

Comment:
can be in future PR because pre-existing: general comment: variables only used once shouldn't be defined in such tests, eg this would be simpler:
```nim
let temperatures = @[-272.15, -2.0, 24.5, 44.31, 99.9, -113.44]
assert temperatures.filterIt(it < 50 and it > -10) == @[-2.0, 24.5, 44.31]
assert temperatures.filterIt(it > 50 or it < -10) == @[-272.15, 99.9, -113.44]
```